### PR TITLE
refactor utf.d: switch to native D types

### PR DESCRIPTION
**Pull request by WalterBright**
trivial

### CODE REVIEW

**Comment by andralex** [comment]
Auto-merge toggled on

File: src/doc.d [hunk 1]
```diff
@@ -2683,13 +2683,13 @@ extern (C++) bool isCVariadicArg(const(char)* p, size_t len)
  */
 extern (C++) bool isIdStart(const(char)* p)
 {
-    uint c = *p;
+    dchar c = *p;
```
Comment (ZombineDev) [inline src/doc.d]: Is this cast correct on big endian systems?

**Comment by ZombineDev** [comment]
@WalterBright the changes are trivial and LGTM.
However I noticed that in some places the following:
```
const(char)* p;
dchar d = *p;
```
This works ok on X86, but would it be correct if the DMD was compiled for some big endian system?

**Comment by WalterBright** [comment]
Yes, because it's still reading just one byte.

**Pull request merged**